Python/Product/Analysis/EncodedLocation.cs

```diff
@@ -39,16 +39,16 @@ struct EncodedLocation : IEquatable<EncodedLocation> {
         public readonly object Location;
 
         public EncodedLocation(ILocationResolver resolver, object location) {
+            if (resolver == null && !(location is LocationInfo)) {
+                throw new ArgumentNullException(nameof(resolver));
+            }
+
             Resolver = resolver;
             Location = location;
         }
 
         public override int GetHashCode() {
-            if (Location != null) {
-                return Resolver.GetHashCode() ^ Location.GetHashCode();
-            }
-
-            return Resolver.GetHashCode();
+            return (Resolver?.GetHashCode() ?? 0) ^ (Location?.GetHashCode() ?? 0);
```

Comment:
Yeah, we can't do 4.7 yet, but I'd be surprised if VS doesn't already include that dependency. Still, not worth it yet just for this one off.